src/net/thread.rs

```diff
@@ -24,13 +28,36 @@ pub enum Message {
 impl Thread {
     #[must_use]
     pub fn new() -> Thread {
+        Thread {
+            thread: Pool::new(1),
+        }
+    }
+
+    /// Sends a function to the running thread for it to be executed.
+    /// # Errors
+    ///  If the thread has been terminated.
+    pub fn execute<F>(&self, f: F) -> Res<()>
+    where
+        F: FnOnce() + Send + 'static,
+    {
+        self.thread.execute(f)?;
+
+        Ok(())
+    }
+}
```

Comment:
There's a nasty trick you can use here...  You can implement Deref on Thread and have it return Pool.  Then any functions on the pool are automatically available when you hold a reference to Thread.

That said, I think that you can probably just delete Thread.  `Pool::new(1)` is shorter to type than `Thread::new()`